### ISSUE TITLE
Rule proposal: `no-useless-spread`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1395]
It's not needed to spread an array literal in array literal or arguments list.

## Fail

```js
const array = [...[foo, ...bar, baz, ]];
```

```js
call(...[foo, ...bar, baz, ]);
```

## Pass

```js
const array = [foo, ...bar, baz,];
```

```js
call(foo, ...bar, baz);
```

**Comment by sindresorhus** [comment]
How about `no-useless-spread` for the name to follow the ESLint naming convention?

**Comment by fisker** [comment]
Should work for object literal too

```js
const object = {
	... {
	    foo,
		...bar,
		baz,
	}
};
```

**Comment by fisker** [comment]
This rule is simple to implement,

The auto fix logic:

1. Remove `...`,  `[`, `]` (or  `{` and `}`) token
2. Remove possible parentheses around 
2. Remove possible trailing comma
3. If it's arguments list, replace holes in the array
